libpod/runtime_ctr.go

```diff
@@ -460,6 +460,25 @@ func (r *Runtime) setupContainer(ctx context.Context, ctr *Container) (_ *Contai
 	} else if err := r.state.AddContainer(ctr); err != nil {
 		return nil, err
 	}
+
+	// Prepare container for volume copy-up if user opted to.
+	if r.config.Containers.PrepareOnCreate {
+		// A data volume container may not have executable file.
+		// Calling init() will result in failure due to executable file
+		// check. Instead, call prepare() to avoid the failure.
+		defer func() {
+			if retErr != nil {
+				if err := ctr.cleanup(ctx); err != nil {
+					logrus.Errorf("error cleaning up container %s: %v", ctr.ID(), err)
```

Comment:
we should probably return this error instead of simply logging it.

Reply:
This is happening in a `defer` call where the dependency is on the function return error `retErr`. This piece of code handling is similar to how is handled elsewhere. I got this code from `libpod/container_internal.go`
```
 249 func (c *Container) handleRestartPolicy(ctx context.Context) (_ bool, retErr error) {
        ...
 277         defer func() {
 278                 if retErr != nil {
 279                         if err := c.cleanup(ctx); err != nil {
 280                                 logrus.Errorf("error cleaning up container %s: %v", c.ID(), err)
 281                         }
 282                 }
 283         }()
 284         if err := c.prepare(); err != nil {
 285                 return false, err
 286         }
 287
```